### DISCUSSION TITLE
Implementing bold+italics inline / how to pass class information for inline elements to rinohtype

### DISCUSSION THREAD

**Discussion by th0mr** [brechtm/rinohtype#412] [Q&A] [answered]
Hi,

RST does not support the combination of bold and italics, unlike markdown which supports it using the triple asterisk ***

After reading this stack overflow answer I've attempted to implement a similar solution using Sphinx+rinohtype https://stackoverflow.com/a/22885876

To allow me to use this in my documents I have added the following to my stylesheet

```
[bold and italic Paragraph : ++Paragraph(has_class='bolditalic')]
base=NEXT_MATCH
font_weight=semi-bold
font_slant=italic

[bold and italic SingleStyledText : ++SingleStyledText(has_class='bolditalic')]
base=NEXT_MATCH
font_weight=semi-bold
font_slant=italic

[bold and italic MixedStyledText : ++MixedStyledText(has_class='bolditalic')]
base=NEXT_MATCH
font_weight=semi-bold
font_slant=italic
```

I then built the following section of document 
```

.. role:: bolditalic
  :class: bolditalic

this is a bold and italics :bolditalic:`attempt`

.. rst-class:: bolditalic

  This paragraph too is bold italic.

```

The paragraphs work fine and are styled with bold and italics, but the inline bolditalics does not.

Here is the stylelog. You can see that the class information is not applied to the inline element.

```
            Paragraph('this is bold and italics attempt...')   C:13 (Path/to/01_introduction.rst)   <paragraph>
                 > (0,0,0,0,2) body [Sphinx] > default
                MixedStyledText('this is a bold and italics attempt...')
                  MixedStyledText('this is a bold and italics attempt...')
                    SingleStyledText('this is a bold and italics ')
                    MixedStyledText('attempt')   <inline>
                      SingleStyledText('attempt')
            Paragraph('This paragraph too is is bold it...')   C:22 (Path/to/01_introduction.rst)   <paragraph>
                 > (2,0,0,1,2) bold and italic Paragraph [mystylesheet] > default
                   (0,0,0,0,2) body [Sphinx] > default
                MixedStyledText('This paragraph too is bold it...')
                  MixedStyledText('This paragraph too is bold it...')
                    SingleStyledText('This paragraph too is bold it...')
```

So, I am a bit lost when it comes to this, it seems I can use the rst-class directive to pass on class information from the rst and then apply the style with rinohtype, but somewhere between the rst -> sphinx -> rinoh process I lose the class information for inline elements when using the role directive.

Any information would be greatly appreciated as I intend to style other inline elements with classes.

**Comment by brechtm** [comment, marked as answer]
I can't remember why the `SingleStyledText` and `MixedStyledText` selectors don't work, but the more generic `StyledText` selector should definitely work. See [`hide.rts`](https://github.com/brechtm/rinohtype/blob/a9268d6209e54038ca5a300bacac5af8d848c29b/tests_regression/rst/hide.rts#L12) (and [`hide.rst`](https://github.com/brechtm/rinohtype/blob/a9268d6209e54038ca5a300bacac5af8d848c29b/tests_regression/rst/hide.rst)) from the regression tests.

Using `SingleStyledText` or `MixedStyledText` should probably throw an error, since I don't think there are actual use cases for it.

**Comment by th0mr** [comment]
Thank you! This solved my problem 👍
